src/bin/commands/demux.rs

```diff
@@ -257,13 +260,7 @@ impl Iterator for ReadSetIterator {
             for read_segment in self.read_structure.iter() {
                 let (seq, quals) = read_segment
                     .extract_bases_and_quals(next_fq_rec.seq(), next_fq_rec.qual())
-                    .unwrap_or_else(|e| {
-                        panic!(
-                            "Error extracting read segment bases or quals from FASTQ record {}; {}",
-                            String::from_utf8(next_fq_rec.head().to_vec()).unwrap(),
-                            e
-                        )
-                    });
+                    .unwrap_or((&EMPTY_BASES, &EMPTY_QUALS));
```

Comment:
I don't like doing this here because it will also eat any other errors coming out of `extract_bases_and_quals`.  I think this will also (unexpectedly) lead to situations where a segment gets returned with 0 bases if it is defined as 8bp long and there are only 7bp available?

Should we perhaps change `.extract_bases_and_quals()` to extract the sub-set of bases that are available, and update the documentation to match?